fluss-server/src/main/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManager.java

```diff
@@ -0,0 +1,223 @@
+                        r -> {
+                            Thread thread = new Thread(r, "rocksdb-metrics-collector");
+                            thread.setDaemon(true);
+                            return thread;
+                        });
+    }
+
+    /** Get the singleton instance of RocksDBMetricsCollector. */
+    public static RocksDBMetricsManager getInstance() {
+        if (instance == null) {
+            synchronized (RocksDBMetricsManager.class) {
+                if (instance == null) {
+                    instance = new RocksDBMetricsManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Register a metrics collector for collection.
+     *
+     * @param collector the collector to register
+     */
+    public void registerCollector(RocksDBMetricsCollector collector) {
+        if (shutdown.get()) {
+            LOG.warn("Cannot register collector after collector is shutdown");
+            return;
+        }
+
+        String key = createCollectorKey(collector);
+        collectors.put(key, collector);
+
+        // Start the collection if this is the first collector
+        if (collectors.size() == 1 && !running.get()) {
+            startCollection();
+        }
+
+        LOG.debug(
+                "Registered RocksDB metrics collector for {}, total collectors: {}",
+                key,
+                collectors.size());
+    }
+
+    /**
+     * Unregister a metrics collector.
+     *
+     * @param collector the collector to unregister
+     */
+    public void unregisterCollector(RocksDBMetricsCollector collector) {
+        String key = createCollectorKey(collector);
+        collectors.remove(key);
+
+        LOG.debug(
+                "Unregistered RocksDB metrics collector for {}, remaining collectors: {}",
+                key,
+                collectors.size());
+    }
+
+    /** Start the metrics collection if not already running. */
+    private void startCollection() {
+        if (running.compareAndSet(false, true)) {
+            LOG.info(
+                    "Starting RocksDB metrics collection with {} threads",
+                    METRICS_COLLECTION_THREADS);
+
+            metricsCollectionExecutor.scheduleAtFixedRate(
+                    this::gatherMetrics,
+                    METRICS_UPDATE_INTERVAL_SECONDS,
+                    METRICS_UPDATE_INTERVAL_SECONDS,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    /** Collect metrics from all registered collectors in batches. */
+    private void gatherMetrics() {
+        if (collectors.isEmpty()) {
+            return;
+        }
+
+        List<RocksDBMetricsCollector> collectorList = new ArrayList<>(collectors.values());
+
+        // Process collectors in batches to avoid overwhelming the system
+        for (int i = 0; i < collectorList.size(); i += BATCH_SIZE) {
+            int endIndex = Math.min(i + BATCH_SIZE, collectorList.size());
```

Comment:
If the schduler is single-thread, then there is no need to submit batching tasks.